lib/openxml/pptx/parts/presentation.rb

```diff
@@ -1,101 +1,147 @@
-          self.slide_id_list = OpenXml::Pptx::Elements::SlideIdList.new
-          self.masters = OpenXml::Pptx::Elements::SlideMasterIdList.new
+        include OpenXml::RelatablePart
+        include OpenXml::HasAttributes
+        include OpenXml::ContainsProperties
+        attr_reader :slides, :slide_masters, :themes
+
+        relationship_type "http://schemas.openxmlformats.org/officeDocument/2006/relationships/officeDocument"
+        content_type "application/vnd.openxmlformats-officedocument.presentationml.presentation.main+xml"
+        default_path "ppt/presentation.xml"
+
+        attribute :auto_compress_pictures, displays_as: :autoCompressPict, expects: :boolean
+        attribute :bookmark_id_seed, displays_as: :bookmarkIdSeed, expects: :integer, in_range: (1..2_147_483_648)
+        attribute :compatibility_mode, displays_as: :compatMode, expects: :boolean
+        attribute :conformance, one_of: %i{ strict transitional }
+        attribute :embed_fonts, displays_as: :embedTrueTypeFonts, expects: :boolean
+        attribute :first_slide_number, displays_as: :firstSlideNum, expects: :integer
+        attribute :remove_personal_information_on_save, displays_as: :removePersonalInfoOnSave, expects: :boolean
+        attribute :right_to_left, displays_as: :rtl, expects: :boolean
+        attribute :save_subset_fonts, displays_as: :saveSubsetFonts, expects: :boolean
+        attribute :server_zoom, displays_as: :serverZoom, expects: :percentage
+        attribute :show_special_placeholders_on_title_slide, displays_as: :showSpecialPlsOnTitleSld, expects: :boolean
+        attribute :strict_first_and_last_characters, displays_as: :strictFirstAndLastChars, expects: :boolean
+
+        property :slide_master_id_list
+        property :notes_master_id_list
+        property :handout_master_id_list
+        property :slide_id_list
+        property :slide_size
+        property :notes_size, required: true
+        property :smart_tags
+        property :embedded_font_list
+        property :custom_show_list
+        property :photo_album
+        # property :customer_data_list # tag :custDataLst
+        # property :kinsoku
+        property :default_text_style
+        # property :modification_verifier # tag :modifyVerifier
+
+        property :extension_list
+
+        # For proper interaction with HasAttributes
+        def self.tag
+          :presentation
         end
 
-        def add_relationship(relationship)
-          relationships.push(relationship)
+        def self.namespace
+          :p
         end
 
-        def add_slide(slide)
-          self.slides.push(slide)
-          slide.add_to(self.slides.size, [self, package])
+        def name
+          "presentation"
         end
 
-        def has_part?(part)
-          package.has_part?(part)
+        def initialize(*_args)
+          super
+          @slides = []
+          @slide_masters = []
+          @themes = []
+
+          presentation_properties
```

Comment:
Is this necessary? — since it's instantiated on demand?

Or ... if it is necessary, does it just make more sense to initialize it once in `initialize` and not to use the on-demand / memoize pattern?

```ruby
@presentation_properties = OpenXml::Pptx::Parts::PresentationProperties.new
add_child_part presentation_properties
```

Reply:
It _is_ required, so yeah, it definitely makes sense to instantiate it in the initializer instead of the on-demand approach.